corehq/apps/es/migration_operations.py

```diff
@@ -110,6 +110,42 @@ def render_index_metadata(type_, mapping, analysis, settings_key, comment):
         }
 
 
+class CreateIndexIfNotExists(CreateIndex):
```

Comment:
This class should live inside the migration itself because this is the only migration that will ever use it (the design).